src/plugins/intel_npu/src/plugin/npuw/weights_bank.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+
+#include "openvino/openvino.hpp"
+#include "openvino/runtime/intel_npu/level_zero/level_zero.hpp"
+#include "openvino/runtime/iremote_context.hpp"
+#include "openvino/runtime/itensor.hpp"
+#include "openvino/runtime/make_tensor.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "plugin.hpp"
+
+namespace ov {
+namespace npuw {
+namespace weights_bank {
+
+class WeightBank {
+public:
+    WeightBank(const std::string& device_name, const std::shared_ptr<const ov::IPlugin>& plugin)
+        : m_device_name(device_name),
+          m_plugin(plugin) {
+        if (m_device_name != "CPU" && m_device_name != "NPU") {
+            OPENVINO_THROW("NPUW doesn't support ", m_device_name, " device for weights sharing!");
+        }
+        if (m_device_name == "NPU" && !m_remote_ctx) {
+            m_remote_ctx = m_plugin->get_core()->get_default_context("NPU")._ptr;
+        }
```

Comment:
Let's drop the NPU & remote tensor specifics here to simplify testing, we need to fix the flow first and having those details here will only make things harder. Let's make one step at a time.

Reply:
Done for now. Planning to move back in this same PR